src/main/java/org/assertj/core/error/ShouldContainExactly.java

```diff
@@ -92,36 +105,49 @@ public static ErrorMessageFactory shouldContainExactly(Object actual, Object exp
 
   public static ErrorMessageFactory shouldHaveSameSize(Object actual, Object expected, int actualSize,
                                                        int expectedSize, ComparisonStrategy comparisonStrategy) {
-    return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize) :
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize, comparisonStrategy);
+
+    Object[] actualArray = arrayOrIterableToArray(actual);
+    Object[] expectedArray = arrayOrIterableToArray(expected);
+    try {
+      List<Delta<String>> diffs = DIFF.diff(actualArray, expectedArray);
+      final String diffString = diffsAsString(diffs);
+      return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
+        new ShouldContainExactly(diffString, actualSize, expectedSize) :
+        new ShouldContainExactly(diffString, actualSize, expectedSize, comparisonStrategy);
+    } catch (IOException e) {
+      throw new RuntimeIOException(format("Unable to verify text contents of path:<%s>", actual), e);
+    }
   }
 
-  private ShouldContainExactly(Object actual, Object expected, int actualSize, int expectedSize,
+  private static Object[] arrayOrIterableToArray(Object input) {
+    if (isArray(input)) {
+      // Convert to List and back to cope with primative arrays.
+      return toArray(Arrays.asList(input));
+    } else {
+      return toArray((Iterable<?>) input);
```

Comment:
are we sure that we will always have an `Iterable` here? 
If is safer to check that input is one and throw an `IllegalArgumentException` saying that it expected an array or an iterable but got something else.